dirty_cat/tests/test_target_encoder.py

```diff
@@ -107,38 +107,92 @@ def test_target_encoder():
         }
 
     ans = np.zeros((n, 2 * 3))
-    Xtest1 = np.array(
-        ["Red", "red", "blue", "green", "Red", "red", "blue", "green"]
-    ).reshape(-1, 1)
-    Xtest2 = np.array(
-        ["male", "male", "male", "male", "female", "female", "female", "female"]
-    ).reshape(-1, 1)
-    Xtest = np.hstack([Xtest1, Xtest2])
     for k in [0, 1, 2]:
         for j, col in enumerate(["color", "gender"]):
             for i in range(n):
-                ans[i, 2 * j + k] = ans_dict[k][col][Xtest[i, j]]
-
+                ans[i, j * 3 + k] = ans_dict[k][col][Xtest[i, j]]
+    encoder = _target_encoder.TargetEncoder(clf_type="multiclass-clf")
+    encoder.fit(X, y)
     Xout = encoder.transform(Xtest)
-    ans = np.zeros((n, 2 * 3))
+    assert np.array_equal(Xout, ans)
+
+
+def test_cv_target_encoder():
+    X1 = np.array(
+        ["Red", "red", "green", "blue", "green", "green", "blue", "red"]
+    ).reshape(-1, 1)
+    X2 = np.array(
+        ["male", "male", "female", "male", "female", "female", "female", "male"]
+    ).reshape(-1, 1)
+    X = np.hstack([X1, X2])
+
+    # Case 1: binary-classification and regression
+    y = np.array([1, 0, 0, 1, 0, 1, 0, 0])
+
+    encoder = _target_encoder.TargetEncoder(cross_val=True)
+    encoder.fit(X, y)
+    t = encoder.transform(X)
+    ft = encoder.fit_transform(X, y)
+    assert np.all(t == ft)
+
+    count_ = {
+        "color": {"Red": 1, "red": 2, "green": 3, "blue": 2},
+        "gender": {"male": 4, "female": 4},
+    }
+    assert count_["color"] == encoder.counter_[0]
+    assert count_["gender"] == encoder.counter_[1]
+
+    for j in range(X.shape[1]):
+        assert np.array_equal(encoder.categories_[j], np.unique(X[:, j]))
+
     Xtest1 = np.array(
         ["Red", "red", "blue", "green", "Red", "red", "blue", "green"]
     ).reshape(-1, 1)
     Xtest2 = np.array(
         ["male", "male", "male", "male", "female", "female", "female", "female"]
     ).reshape(-1, 1)
     Xtest = np.hstack([Xtest1, Xtest2])
-    for k in [0, 1, 2]:
-        for j, col in enumerate(["color", "gender"]):
-            for i in range(n):
-                ans[i, j * 3 + k] = ans_dict[k][col][Xtest[i, j]]
-    encoder = _target_encoder.TargetEncoder(clf_type="multiclass-clf")
+
+    Xout = encoder.transform(Xtest)
+
+    # Test same seed returns the same results:
+    encoder2 = _target_encoder.TargetEncoder(cross_val=True)
+    encoder2.fit(X, y)
```

Comment:
So, this seems to answers my question above, but checking the `cross_val` documentation, it is not mentioned. Maybe adding a note could avoid some confusion!